packages/engine/Source/Core/JulianDate.js

```diff
@@ -783,40 +783,16 @@ JulianDate.toIso8601 = function (julianDate, precision) {
 
   let millisecondStr;
 
-  if (!defined(precision) && millisecond !== 0) {
-    //Forces milliseconds into a number with at least 3 digits to whatever the default toString() precision is.
-    millisecondStr = (millisecond * 0.01).toString().replace(".", "");
-    return `${year.toString().padStart(4, "0")}-${month
-      .toString()
-      .padStart(2, "0")}-${day
-      .toString()
-      .padStart(2, "0")}T${hour
-      .toString()
-      .padStart(2, "0")}:${minute
-      .toString()
-      .padStart(2, "0")}:${second
-      .toString()
-      .padStart(2, "0")}.${millisecondStr}Z`;
-  }
-
-  //Precision is either 0 or milliseconds is 0 with undefined precision, in either case, leave off milliseconds entirely
-  if (!defined(precision) || precision === 0) {
-    return `${year.toString().padStart(4, "0")}-${month
-      .toString()
-      .padStart(2, "0")}-${day
-      .toString()
-      .padStart(2, "0")}T${hour
-      .toString()
-      .padStart(2, "0")}:${minute
-      .toString()
-      .padStart(2, "0")}:${second.toString().padStart(2, "0")}Z`;
-  }
-
-  //Forces milliseconds into a number with at least 3 digits to whatever the specified precision is.
-  millisecondStr = (millisecond * 0.01)
-    .toFixed(precision)
-    .replace(".", "")
-    .slice(0, precision);
+  if (millisecond !== 0 || defined(precision)) {
+    if (!defined(precision)) {
+      precision = 7; // Default precision if not provided
+    }
```

Comment:
The previous version of this function did _not_ use a default precision. Any revisions should also not use a default precision to avoid breaking existing applications.

Reply:
Removed.